src/REST/keywords.py

```diff
@@ -77,6 +78,45 @@ def set_client_cert(self, cert):
         self.request["cert"] = self._input_client_cert(cert)
         return self.request["cert"]
 
+
+    @keyword(name="Set Client Authentication", tags=("settings",))
+    def set_client_authentication(self, auth_type, user=None, password=None):
+        """*Attaches HTTP basic authentication to the given requests.*
+
+        The API by default will not have authentication enabled. 
+
+        The auth_type argument must be ${NONE}, basic, digest or proxy.
+        In case the user sets the auth_type to ${NONE} the authentication 
+        information 
+        The user and password will be written in plain text.
+
+        *Examples*
+
+        | `Set Client Authentication` | basic  | admin | password |
+        | `Set Client Authentication` | digest | admin | password |
+        | `Set Client Authentication` | proxy  | admin | password |
+        | `Set Client Authentication` | ${NONE} | | |
+        """
+        if auth_type != None and not isinstance(auth_type, str):
+            raise TypeError(
+                    "Argument \"auth_type\" must be string or ${NONE}!"
+            )
+
+        if auth_type != None:
+            auth_type = auth_type.lower()
+            if auth_type == "basic" :
+                auth_type = HTTPBasicAuth
+            elif auth_type == "digest" :
+                auth_type = HTTPDigestAuth
+            elif auth_type == "proxy" :
+                auth_type = HTTPProxyAuth
+            else :
+                raise Exception( 
+                        "Argument \"auth_type\" must be ${NONE}, basic, digest or proxy."
```

Comment:
The earlier error messasge is that `auth_type` must be a string or ${NONE}. I think it would be clearer if the error messages were the same. Also these could be combined into a single try-except.